docker-compose.yml

```diff
@@ -0,0 +1,37 @@
+version: "3"
+
+services:
+  webapp-prod:
+    container_name: production
+    build:
+      dockerfile:  Dockerfile
+      context: .
+      target: production
+    env_file:
+      - .env
+    ports:
+      - 80:80
+  webapp-dev:
+    container_name: development
+    build:
+      dockerfile: Dockerfile
+      context: .
+      target: development
+    volumes:
+      - type: bind
+        source: ./todo_app
+        target: /opt/todo_app
+    env_file:
+      - .env.dev
+    ports:
+      - 5000:5000
+  webapp-test:
+    container_name: test
+    build:
+      dockerfile: Dockerfile
+      context: .
+      target: test
+    volumes:
+      - type: bind
+        source: ./
```

Comment:
This might be okay on a mac, but note that you are mounting the whole of your current directory including the .venv folder. The venv folder from a Windows machine may not work inside the container.

Any reason to mount the whole folder instead of doing the same as your "webapp-dev" service? `source: ./todo_app` etc